web/static/components/transfer/transfer-endpoint-manager.js

```diff
@@ -0,0 +1,133 @@
+import * as api from "../../api.js";
+import * as dialogs from "../../dialogs.js";
+
+export class TransferEndpointManager {
+    constructor(dialog) {
+        this.initialized = false;
+        this.controller = dialog;
+        this.currentEndpoint = null;
+        this.endpointManagerList = null;
+        this.searchCounter = 0;
+        this.currentSearchToken = null;
+    }
+
+    /**
+     * ------------GET------------
+     */
+
+    getEndpointStatus(endpoint) {
+        if (!endpoint.activated && endpoint.expires_in === -1) return "active";
```

Comment:
I also prefer { } around content in if statements. This prevents accidental newlines from introducing errors.